### PR TITLE
Feature/i delet user user service

### PR DESCRIPTION
**Pull request by Dimentr01**

### CODE REVIEW

File: Services/UserService/Commands/DeleteUserCommand.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿ using System;
+ using System.Net;
+ using IDeleteUserUserService.Interfaces;
+
+
+ namespace UserService.Commands
+
```
Comment (nmamro) [inline Services/UserService/Commands/DeleteUserCommand.cs]: What is this empty line?

File: Services/UserService/Controllers/UserServiceController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
```
Comment (nmamro) [inline Services/UserService/Controllers/UserServiceController.cs]: Are all these using used?

File: Services/UserService/Controllers/UserServiceController.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using System.Threading.Tasks;
+using IDeleteUserUserService.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+using UserService.Commands;
+
+namespace UserService.Controllers
+{
+    [ApiController]
+
```
Comment (nmamro) [inline Services/UserService/Controllers/UserServiceController.cs]: Why this empty line is here?

File: Services/UserService/Controllers/UserServiceController.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public class UserServiceController : ControllerBase
+    {
+        [Route("[controller]/deleteUser")]
+        [HttpGet]
+        public HttpStatusCode DeleteUserFromDataBaseService([FromServices] IDeleteUser command,[FromQuery] int userId)
```
Comment (nmamro) [inline Services/UserService/Controllers/UserServiceController.cs]: What does deleting user from service mean?

File: Services/UserService/Interfaces/IDeleteUser.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Net;
+
+namespace IDeleteUserUserService.Interfaces
+{
+
```
Comment (nmamro) [inline Services/UserService/Interfaces/IDeleteUser.cs]: I don't know, mb it's a bug, but these empty lines look strange

**Review by nmamro — DISMISSED** [review]
It can look a bit better

**Comment by nmamro** [comment]
Why it's "delet" hm

**Review by Maksim-Trolina — APPROVED** [review]

File: Services/UserService/Interfaces/IDeleteUser.cs (hunk 5 above)
Comment (Dimentr01) [reply Services/UserService/Interfaces/IDeleteUser.cs]: > I don't know, mb it's a bug, but these empty lines look strange

it is just empty; I wrote the code so

File: Services/UserService/Controllers/UserServiceController.cs (hunk 3 above)
Comment (Dimentr01) [reply Services/UserService/Controllers/UserServiceController.cs]: > Why this empty line is here?

I deleted the lines and started writing new ones, so there was an empty line

File: Services/UserService/Commands/DeleteUserCommand.cs (hunk 1 above)
Comment (Dimentr01) [reply Services/UserService/Commands/DeleteUserCommand.cs]: > What is this empty line?

I deleted the lines and started writing new ones, so there was an empty line

**Comment by Dimentr01** [comment]
> It can look a bit better

maybe

File: Services/UserService/Controllers/UserServiceController.cs (hunk 4 above)
Comment (Dimentr01) [reply Services/UserService/Controllers/UserServiceController.cs]: > What does deleting user from service mean?

it's the Route

File: Services/UserService/Controllers/UserServiceController.cs (hunk 2 above)
Comment (Dimentr01) [reply Services/UserService/Controllers/UserServiceController.cs]: > Are all these using used?

No, my mistake

File: Services/UserService/Interfaces/IDeleteUser.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    public interface IDeleteUser
```
Comment (DarkFelas) [inline Services/UserService/Interfaces/IDeleteUser.cs]: It is a good idea to name such interface like IDeleteUser**Command**

File: Services/UserService/Interfaces/IDeleteUser.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    {
+        HttpStatusCode Execute(int userId);
```
Comment (DarkFelas) [inline Services/UserService/Interfaces/IDeleteUser.cs]: No need to return HttpStatusCode, **void** is good here and if you want something more you can use **ActionResult**

File: Services/UserService/Commands/DeleteUserCommand.cs (hunk 1 above)
Comment (DarkFelas) [reply Services/UserService/Commands/DeleteUserCommand.cs]: Remove blank line

File: Services/UserService/Commands/DeleteUserCommand.cs [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ {
+     public class DeleteUserCommand : IDeleteUser
+     {
+         public HttpStatusCode Execute(int userId)
+         {
+             return DeleteUserFromDataBaseService(userId) == "Ok" ? HttpStatusCode.OK : HttpStatusCode.BadRequest;
+         }
+
+        //TODO fix from stub on request
+         private string DeleteUserFromDataBaseService(int userId)
```
Comment (DarkFelas) [inline Services/UserService/Commands/DeleteUserCommand.cs]: Not sure about **FromDataBaseService**, imho it is not good name and why this method return something?

File: Services/UserService/Controllers/UserServiceController.cs (hunk 3 above)
Comment (DarkFelas) [reply Services/UserService/Controllers/UserServiceController.cs]: remove blank line

File: Services/UserService/Controllers/UserServiceController.cs (hunk 4 above)
Comment (DarkFelas) [reply Services/UserService/Controllers/UserServiceController.cs]: agreed with Nikita it is a bad name for controller endpoint, **DeleteUser** will be enough
Check userId type, i'm not sure about **int**

**Review by DarkFelas — DISMISSED** [review]
need more work

File: Services/UserService/Commands/DeleteUserCommand.cs (hunk 8 above)
Comment (Dimentr01) [reply Services/UserService/Commands/DeleteUserCommand.cs]: > Not sure about **FromDataBaseService**, imho it is not good name and why this method return something?

it is the stub

**Pull request merged**